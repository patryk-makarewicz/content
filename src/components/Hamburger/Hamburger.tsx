import { useEffect } from 'react';

import * as Styled from './Hamburger.styles';

export type HamburgerProps = {
  showMenu: boolean;
  onOpenCloseMenu: () => void;
};

export const Hamburger = ({ showMenu, onOpenCloseMenu }: HamburgerProps) => {
  useEffect(() => {
    if (showMenu) {
      document.body.style.overflow = 'hidden';
    } else {
      document.body.style.overflow = '';
    }
  }, [showMenu]);

  return (
    <Styled.HamburgerButton type="button" onClick={onOpenCloseMenu} data-testid="button-hamburger">
      <Styled.BarOne open={showMenu} />
      <Styled.BarTwo open={showMenu} />
      <Styled.BarThree open={showMenu} />
    </Styled.HamburgerButton>
  );
};
