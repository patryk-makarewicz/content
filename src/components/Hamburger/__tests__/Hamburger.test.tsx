import { render } from '@testing-library/react';
import React from 'react';

import { Hamburger, HamburgerProps } from '../Hamburger';

const renderHamburger = ({ showMenu = false }: Partial<HamburgerProps> = {}) => {
  const onOpenCloseMenu = jest.fn();
  const { baseElement } = render(<Hamburger onOpenCloseMenu={onOpenCloseMenu} showMenu={showMenu} />);

  return {
    baseElement,
  };
};

describe('test Hamburger', () => {
  it('should take the snapshot when close', () => {
    const { baseElement } = renderHamburger();

    expect(baseElement).toMatchSnapshot();
  });

  it('should take the snapshot when open', () => {
    const { baseElement } = renderHamburger({ showMenu: true });

    expect(baseElement).toMatchSnapshot();
  });
});
