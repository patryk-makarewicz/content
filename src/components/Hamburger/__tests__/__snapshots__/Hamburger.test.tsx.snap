// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test Hamburger should take the snapshot when close 1`] = `
<body>
  <div>
    <button
      class="sc-aXZVg Gvapt"
      data-testid="button-hamburger"
      type="button"
    >
      <div
        class="sc-gEvEer cfdbZP"
      />
      <div
        class="sc-gEvEer sc-eqUAAy cfdbZP giEXpz"
      />
      <div
        class="sc-gEvEer sc-fqkvVR cfdbZP eJjnVN"
      />
    </button>
  </div>
</body>
`;

exports[`test Hamburger should take the snapshot when open 1`] = `
<body
  style="overflow: hidden;"
>
  <div>
    <button
      class="sc-aXZVg Gvapt"
      data-testid="button-hamburger"
      type="button"
    >
      <div
        class="sc-gEvEer fsOJRY"
        open=""
      />
      <div
        class="sc-gEvEer sc-eqUAAy fsOJRY giJuLz"
        open=""
      />
      <div
        class="sc-gEvEer sc-fqkvVR fsOJRY iCXEHq"
        open=""
      />
    </button>
  </div>
</body>
`;
